src/commands/git/branch.ts

```diff
@@ -516,10 +517,19 @@ export class BranchGitCommand extends QuickCommand {
 			state.flags = result;
 
 			endSteps(state);
-			state.repo.branchDelete(state.references, {
-				force: state.flags.includes('--force'),
-				remote: state.flags.includes('--remotes'),
-			});
+
+			try {
+				await state.repo.git.deleteBranch(state.references, {
+					force: state.flags.includes('--force'),
+					remote: state.flags.includes('--remotes'),
+				});
+			} catch (ex) {
+				Logger.error(ex);
+				// TODO likely need some better error handling here
+				return showGenericErrorMessage(
+					new BranchError(ex.reason, ex, state.references.map(r => r.name).join(', ')).message,
```

Comment:
Why are we creating a BranchError here? Isn't the one being thrown a BranchError?

Reply:
Indeed, I don't really like this solution either. When we return the BranchError at git level we don't know the branch names, we just have an array of args. We could either guess the branch names, or change the number of args of `git.branch` to be more specific so we can have easier access. In this case we're reusing the BranchError but adding the branch name for context.
Another option is to turn BranchError into a Builder pattern that we can always call like `showGenericErrorMessage(ex.WithBranch(branchName))` or something along those lines to provide context on top layers.